src/rosdep2/sources_list.py

```diff
@@ -320,9 +321,12 @@ def download_default_sources_list(url=DEFAULT_SOURCES_LIST_URL):
     data = f.read().decode()
     f.close()
     if not data:
-        raise RuntimeError("cannot download defaults file: empty contents")
+        raise DownloadFailure("cannot download defaults file: empty contents")
     # parse just for validation
-    parse_sources_data(data)
+    try:
+        parse_sources_data(data)
+    except InvalidData as e:
+        raise DownloadFailure("Failed to download valid rosdep sources from %s Contents were:{{{%s}}} Parsing error: %s" % (url, data, e))
```

Comment:
Of the available exceptions this looked the most appropriate to me. If there's one you deem more appropriate I'm happy to switch. 

Although the error may be unrelated to the actual transport, if the downloaded data is invalid or corrupted the place to look is at the network or source. This is following the same pattern as the exceptions in the function `download_rosdep_data` where data downloaded failing validation is considered a failed download. And the DownloadFailed exception is the one that makes sense to add urls to me.